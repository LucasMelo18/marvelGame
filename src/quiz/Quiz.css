.mainQuiz{
  align-items: center;
  display: flex;
  height: 100vh;
  justify-content: center;
}

.mainQuiz header{
  width: 500px;
  height: 90vh;
  border: 1.5px solid #ee4326;
  border-radius: 20px;
}

.mainQuiz header p{
  font-size: 28pt;
  margin: 10px 0 20px 0;
  color: #ee4326;
  font-family: 'Montserrat', sans-serif;
}

.mainQuiz header p:nth-child(3){
  color: rgb(128, 128, 128);
  font-size: 10pt;
  margin: 0;
}

.mainQuiz .logoQuiz{
  width: 100px;
  height: auto;
}

.mainQuiz header h1{
  color: #ee4326;
  font-family: 'Montserrat', sans-serif;
  font-size: 20pt;
  margin-top: 0;
}

.mainQuiz header section img{
  border-radius: 10px;
  width: 200px;
}

.mainQuiz div input[type="text"]{
  border: none;
  border-bottom: 1.5px solid #ee4326;
  color: rgb(128, 128, 128);
  font-family: 'Montserrat', sans-serif;
  margin: 30px 0;
  text-align: center;
  text-transform: uppercase;
  width: 50%;
}

.mainQuiz div #answerButton{
  background-color: #ee4326;
  border-radius: 30px;
  border: none;
  color: #FFF;
  font-size: 12pt;
  height: 40px;
  margin-bottom: 30px;
  width: 30%;
}

.mainQuiz div #answerButton:hover{
  background-color: #ff664b;
  cursor: pointer;
}

.mainQuiz div input[type="text"]::placeholder{
  color: rgb(128, 128, 128);
  border-bottom: 2px solid #ee4326;
  width: 50%;
  margin: 30px 0;
}

.mainQuiz #helpButtons{
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 100%;
}

.mainQuiz #helpButtons input[type="submit"]{
  border: none;
  background-color: #1c5de0;
  border-radius: 30px;
  color: #FFF;
  cursor: pointer;
  height: auto;
  width: 150px;
}

@media screen and (max-width: 570px){
  .mainQuiz header{
    width: 400px;
  }
}

@media screen and (max-width: 430px){
  .mainQuiz header{
    width: 300px;
  }
}

@media screen and (max-width: 320px){
  .mainQuiz header{
    width: 280px;
  }
}