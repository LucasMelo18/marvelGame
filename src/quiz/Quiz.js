import React,  { Component } from 'react';
import logoMarvel from '../assets/images/marvelLogo.jpg';
import thorImage from '../assets/images/thorImage.png';
import Overlay from 'react-bootstrap/Overlay'

import './Quiz.css';


class Quiz extends Component {
  constructor(...args){
    super(...args)

    this.attachRef = target => this.setState({ target });
    this.state={
      contador: 0,
      show: false
    }
  }

  toIncreaseContador =() =>{
    this.setState({
      contador: this.state.contador +1,
    });
  }


  toDecrementContador = () =>{
    this.setState({
      contador: this.state.contador -1,
      show: !this.state.show,
    })
  }

  toClearContador = () => {
    this.setState({
      contador: 0,
    })
  }

  render() {
    const { show, target } = this.state;
    return (
      <div className="mainQuiz">
          <header>
            <img className="logoQuiz" src={logoMarvel} alt="Logo quiz Marvel"/>
            <h1>Quiz</h1>
            <p>Sua pontuação é:</p>
            <p>{this.state.contador}</p>

            {/* ÁREA DE GERAÇÃO DAS IMAGENS DO QUIZ */}
            <section>
              <img src={thorImage} alt="Foto THOR"/>
            </section>

            {/* INPUT DE INSERÇÃO DA RESPOSTA */}
            <section>
              <div>
                <input type="text" placeholder="Digite sua resposta"/>
              </div>

              {/* BOTOES DE RESPOSTA AJUDA E DESISTENCIA */}
              <div>
                <button id="answerButton" onClick={this.toIncreaseContador}>Responder</button>
              </div>
              
              <div id="helpButtons">
                <input type="submit" value="Mostrar resposta" onClick={this.toClearContador}/>
                
                <input type="submit" value="Dica" ref={this.attachRef} onClick={this.toDecrementContador}/>
                <Overlay target={target} show={show} placement="right">
                  {({ placement, scheduleUpdate, arrowProps, ...props }) => (
                    <div {...props} style={{ backgroundColor: 'rgb(255, 100, 100)', padding: '2px 10px', color: 'white', borderRadius: 3, ...props.style}}>
                      Deus nórdico, dos trovões e das batalhas
                    </div>
                  )}
                </Overlay>
              </div>
            </section>
          </header>
      </div>
    )
  }
}

export default Quiz;
