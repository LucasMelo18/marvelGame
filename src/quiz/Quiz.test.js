import React from 'react';
import renderer from 'react-test-renderer';

import Quiz from './Quiz';

describe('Quiz', () => {
  var subject;
  var tree;

  beforeEach(() => {
    subject = renderer.create(<Quiz />);
    tree = subject.toJSON();
  });

  it('is expected to match the snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
});
