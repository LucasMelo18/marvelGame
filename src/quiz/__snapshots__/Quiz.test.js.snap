// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz is expected to match the snapshot 1`] = `
<div
  className="mainQuiz"
>
  <header>
    <img
      alt="Logo quiz Marvel"
      className="logoQuiz"
      src="marvelLogo.jpg"
    />
    <h1>
      Quiz
    </h1>
    <p>
      Sua pontuação é:
    </p>
    <p>
      0
    </p>
    <section>
      <img
        alt="Foto THOR"
        src="thorImage.png"
      />
    </section>
    <section>
      <div>
        <input
          placeholder="Digite sua resposta"
          type="text"
        />
      </div>
      <div>
        <button
          id="answerButton"
          onClick={[Function]}
        >
          Responder
        </button>
      </div>
      <div
        id="helpButtons"
      >
        <input
          onClick={[Function]}
          type="submit"
          value="Mostrar resposta"
        />
        <input
          onClick={[Function]}
          type="submit"
          value="Dica"
        />
      </div>
    </section>
  </header>
</div>
`;
